From 2dfe66886d71b9a341433ea8b6ff225cc07da80b Mon Sep 17 00:00:00 2001
From: Andrey Semashev <andrey.semashev@gmail.com>
Date: Mon, 2 Dec 2024 19:22:20 +0300
Subject: Add a comma before ellipsis in constexpr_swap.

gcc 15 complains that the comma is required before vararg ellipsis.

Fixes https://github.com/boostorg/integer/issues/35.
---
 include/boost/integer/common_factor_rt.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/boost/integer/common_factor_rt.hpp b/include/boost/integer/common_factor_rt.hpp
index d6765d4..817682f 100644
--- a/include/boost/integer/common_factor_rt.hpp
+++ b/include/boost/integer/common_factor_rt.hpp
@@ -64,7 +64,7 @@ namespace boost {
             return a.swap(b);
          }
          template <class T, class U>
-         inline constexpr void constexpr_swap(T&a, U& b...) BOOST_GCD_NOEXCEPT(T)
+         inline constexpr void constexpr_swap(T& a, U& b, ...) BOOST_GCD_NOEXCEPT(T)
          {
             T t(static_cast<T&&>(a));
             a = static_cast<T&&>(b);
-- 
2.47.1

